### ISSUE TITLE
Update deprecated use of Puppeteer's `page.waitFor`

### ISSUE THREAD

**Issue by aaemnnosttv** [google/site-kit-wp#3236]
## Feature Description

v5 of Puppeteer has deprecated its `waitFor` function. We should update our use accordingly to avoid problems later when it is removed, but also to avoid extra noise in tests from this warning.

This function is still not removed yet, even in the latest v9 of Puppeteer so it isn't an emergency.

See also https://github.com/WordPress/gutenberg/issues/28093

---------------

_Do not alter or remove anything below. The following sections will be managed by moderators only._

## Acceptance criteria

* All uses of `page.waitFor` in E2E tests should be updated to use `page.waitForTimeout` instead  
See https://pptr.dev/#?product=Puppeteer&version=v9.0.0&show=api-pagewaitforselectororfunctionortimeout-options-args
* The same change should also be made in our BackstopJS helpers/utilities  
E.g. https://github.com/google/site-kit-wp/blob/3fb4b73421597f329da0ce0022ba315a6cf18630/tests/backstop/engine_scripts/puppet/clickAndHoverHelper.js#L10

## Implementation Brief

* Upgrade Puppeteer in `package.json` so it includes these new methods
* Replace `page.waitFor` https://github.com/google/site-kit-wp/blob/2a18934d46d4cadcb169c50c406ec13f28c481d6/tests/e2e/utils/page-wait.js#L32 with `page.waitForTimeout`.
* Replace `page.waitFor` with `page.waitForSelector` in https://github.com/google/site-kit-wp/blob/2a18934d46d4cadcb169c50c406ec13f28c481d6/tests/backstop/engine_scripts/puppet/clickAndHoverHelper.js
* While the above covers most of the issue, it looks like a few E2E tests need fixing, see errors here: https://github.com/google/site-kit-wp/pull/3258/checks?check_run_id=2504429348
  * It looks like some requests are no longer being blocked by Puppeteer and are running, causing errors about bad request params or invalid CORS headers, so it seems that our Puppeteer request interception needs to be adjusted.
* See #3258 for the start of this PR—all that remains to be fixed are the few tests failing due to requests being sent to the live servers.

### Test Coverage

* No new tests required.

### Visual Regression Changes

* N/A.

## QA Brief

* E2E tests as well as VRT/BackstopJS should not log deprecation messages about `page.waitFor` calls. You can check the PR or a `develop` GitHub Action run of E2E tests to verify this.

## Changelog entry

* N/A

**Comment by aaemnnosttv** [comment]
> Upgrade Puppeteer in `package.json` so it includes these new methods

I don't think we need to add `puppeteer` as a top-level dependency, in fact, [we just removed it as one in a recent issue](https://github.com/google/site-kit-wp/issues/1797). Did you find that this was necessary? The new method is available since v5.3 so it looks like our version already meets that.

> While the above covers most of the issue, it looks like a few E2E tests need fixing, see errors here

If we do need to upgrade puppeteer first, it might be a good idea to create a new issue for upgrading `@wordpress/scripts` first which would also be good in preparation for #3340 as well.

**Comment by danielgent** [comment]
@aaemnnosttv  There already is an issue for upgrading wp-scripts *and* I'm already doing the IB for it (Sadly github actions being down means I can't continue, but I've got upgrades working up to v14) https://github.com/google/site-kit-wp/issues/3033

**Comment by aaemnnosttv** [comment]
> * Replace `page.waitFor` [...] with `page.waitForDelay`.

@tofumatt `waitForDelay` isn't a valid method, this should be `waitForTimeout` instead.

This looks good to go otherwise so I'll rename this one detail in the IB.

IB ✅

**Comment by techanvil** [comment]
## QA :white_check_mark: 

Confirmed on recent runs of the VRT and E2E tests that no warnings about deprecated `waitFor` calls are being logged.
